src/components/PopupWithConfirmation.js

```diff
@@ -1,39 +1,38 @@
 import Popup from "./Popup.js";
-
 export default class PopupWithConfirmation extends Popup {
   constructor(popupSelector) {
-    super(popupSelector);
-    this._form = this._popup.querySelector(".modal__form");
-    this._submitButton = this._popup.querySelector('.modal__button');
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupForm.querySelector(".modal__button");
     this._submitButtonText = this._submitButton.textContent;
   }
+  // open() {
+  //   this._popupForm.reset();
+  //   super.open();
+  // }
+  // close() {
+  //   this._popupForm.reset();
+  //   super.close();
+  // }
 
- {
-    super();
-  }
-
-  close() {
-    this._form.reset();
-    super.close();
-  }
-
-  setButtonText(submit, buttonText = "Saving...") {
+  setButtonText(submit, buttonText = "Deleting...") {
     if (submit) {
-      console.log(this._submitButton)
       this._submitButton.textContent = buttonText;
     } else {
       this._submitButton.textContent = this._submitButtonText;
     }
   }
-
-  setSubmitAction(callback) {
+  setSubmitCallback(callback) {
```

Comment:
`setSubmitAction`  doesn't exit


Please, return the right code back. I correct one line the other error appears and so on